internal/cmd/testenv/main.go

```diff
@@ -0,0 +1,147 @@
+)
+
+func main() {
+	checkOnly := func() bool {
+		cleanup := flag.Bool("cleanup", false, "Perform cleanup tasks post CI")
+		precheck := flag.Bool("precheck", false, "Check pre-requirements for running CI tests")
+		flag.Parse()
+		if (!*cleanup && !*precheck) ||
+			(*cleanup && *precheck) {
+			flag.Usage()
+			os.Exit(1)
+		}
+		return *precheck
+	}()
+
+	ctx := context.Background()
+
+	// Load configuration and create client
+	polAccount, err := polaris.DefaultServiceAccount(true)
+	if err != nil {
+		log.Fatal(err)
+	}
+	client, err := polaris.NewClient(ctx, polAccount, &polaris_log.StandardLogger{})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var g errgroup.Group
+
+	// AWS
+	g.Go(func() error {
+		awsAccount, err := client.AWS().Account(ctx, aws.ID(aws.Default()), core.FeatureAll)
+		if checkOnly {
+			switch {
+			case err == nil:
+				return fmt.Errorf("found pre-existing AWS account: %v", pretty.Sprint(awsAccount))
+			case !errors.Is(err, graphql.ErrNotFound):
+				return fmt.Errorf("failed to check AWS account: %v", err)
+			}
+			return nil
+		}
+
+		// cleanup phase
+		switch {
+		case errors.Is(err, graphql.ErrNotFound):
+			return nil
+		case err != nil:
+			return fmt.Errorf("failed to check AWS account: %v", err)
+		}
+		testAcc, err := testsetup.AWSAccount()
+		if err != nil {
+			log.Fatal(err)
+		}
+		if awsAccount.NativeID != testAcc.AccountID {
+			return fmt.Errorf("existing AWS account %q isn't expected test account %q, won't remove",
+				awsAccount.NativeID, testAcc.AccountID)
+		}
+		// TODO: we might need to iterate over awsAccount.Features to remove
+		// all of them in the future
+		return client.AWS().RemoveAccount(ctx, aws.Default(), core.FeatureCloudNativeProtection, false)
+	})
+
+	// Azure
+	g.Go(func() error {
+		azureAccs, err := client.Azure().Subscriptions(ctx, core.FeatureAll, "")
+		if err != nil {
+			return err
+		}
+		if len(azureAccs) == 0 {
+			return nil
+		}
+		if checkOnly {
+			return fmt.Errorf("found pre-existing Azure subscriptions: %v", pretty.Sprint(azureAccs))
+		}
+
+		// cleanup phase
+		if len(azureAccs) > 1 {
+			return fmt.Errorf("expected one Azure account, found: %v", pretty.Sprint(azureAccs))
+		}
+		// TODO: do we need to check ExocomputeConfigs / NativeSubscriptions as
+		// well?
+		testSub, err := testsetup.AzureSubscription()
+		if err != nil {
+			return err
+		}
+		if azureAccs[0].NativeID != testSub.SubscriptionID {
+			return fmt.Errorf("existing Azure subscription %q isn't the expected test subscription %q, won't remove",
+				azureAccs[0].NativeID, testSub.SubscriptionID)
+		}
+		sub := azure.Subscription(testSub.SubscriptionID, testSub.TenantDomain)
+		// TODO: we might need to iterate over azureAccs[0].Features to remove
+		// all of them in the future
+		return client.Azure().RemoveSubscription(ctx, azure.ID(sub), core.FeatureCloudNativeProtection, false)
+	})
+
+	// GCP
+	g.Go(func() error {
+		projs, err := client.GCP().Projects(ctx, core.FeatureAll, "")
```

Comment:
Is there a reason to not ask for the specific subscription? Something like this:
```
testProj, err := testsetup.GCPProject()

...

projs, err := client.GCP().Project(ctx, gcp.ProjectID(testProj.ProjectID), core.FeatureCloudNativeProtection)
```